tools/checkcomments/checkcomments.go

```diff
@@ -72,45 +124,75 @@ func run(pass *analysis.Pass) (any, error) {
 					continue
 				}
 
-				// `go vet -vettool` runs checkcomments once per package.
-				// That causes same issues to be checked multiple times,
-				// and that easily pushes us over the rate limit.
-				// We should cache check results using lockedfile.
-				// TODO https://github.com/FerretDB/FerretDB/issues/2733
+				iNum := match[1]
 
-				/*
-					n, err := strconv.Atoi(match[1])
-					if err != nil {
-						log.Fatal(err)
+				_, ok := iCache.Issues[iNum]
+
+				if !ok {
+					n, inErr := strconv.Atoi(iNum)
+					if inErr != nil {
+						log.Fatal(inErr)
 					}
 
-					open, ok := issues[n]
-					if !ok {
-						issue, _, err := client.Issues.Get(context.TODO(), "FerretDB", "FerretDB", n)
-						if err != nil {
-							if errors.As(err, new(*github.RateLimitError)) && token == "" {
-								log.Println(
-									"Rate limit reached. Please set a GITHUB_TOKEN as described at",
-									"https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#setting-a-github_token",
-								)
+					isOpen, inErr := isIssueOpen(client, n)
+					if err != nil {
+						log.Fatal(inErr)
+					}
 
-								return nil, nil
-							}
+					iCache.Issues[iNum] = isOpen
+				}
 
-							log.Fatalf("%[1]T %[1]s", err)
-						}
+				if !iCache.Issues[iNum] {
+					message := fmt.Sprintf("invalid TODO: linked issue %s is closed", iNum)
+					pass.Reportf(c.Pos(), message)
+				}
+			}
+		}
+	}
 
-						open = issue.GetState() == "open"
-						issues[n] = open
-					}
+	if len(iCache.Issues) > 0 {
+		jsonb, err := json.Marshal(iCache)
+		if err != nil {
+			log.Fatal(err)
+		}
 
-					if !open {
-						pass.Reportf(c.Pos(), "invalid TODO: linked issue is closed")
-					}
-				*/
-			}
+		_, err = cf.WriteAt(jsonb, 0)
+		if err != nil {
+			log.Fatal(err)
 		}
 	}
 
 	return nil, nil
 }
+
+func isIssueOpen(client *github.Client, n int) (bool, error) {
+	issue, _, err := client.Issues.Get(context.TODO(), "FerretDB", "FerretDB", n)
+	if err != nil {
+		if errors.As(err, new(*github.RateLimitError)) && os.Getenv("GITHUB_TOKEN") == "" {
+			log.Println(
+				"Rate limit reached. Please set a GITHUB_TOKEN as described at",
+				"https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#setting-a-github_token",
+			)
+
+			return false, err
+		}
+	}
+
+	isOpen := issue.GetState() == "open"
+
+	return isOpen, nil
+}
+
+func getCacheFilePath(p string) string {
+	path := filepath.Dir(p)
+
+	readmePath := filepath.Join(path, "README.md")
```

Comment:
This part is not clear to me. Why do we check if README.md exists? Maybe it is worth to leave a comment in the code.

Reply:
@raeidish thanks for the clarification! Yeah, it makes sense. This PR is almost ready to be merged, so I'm just going through all my previous questions at the moment.